lua-iconv-dev-1.rockspec

```diff
@@ -1,22 +1,27 @@
--- Packs lua-iconv into a LuaRock
--- rockspec based uppon the file provided by DarkGod <darkgod at net-core.org>
+local package_name = "lua-iconv"
+local package_version = "dev"
+local rockspec_revision = "1"
+local github_account_name = "lunarmodules"
+local github_repo_name = package_name
 
-package = "lua-iconv"
-version = "dev-1"
+package = package_name
+version = package_version.."-"..rockspec_revision
 
 source = {
-  url = "git+https://github.com/lunarmodules/lua-iconv",
+   url = "git+https://github.com/"..github_account_name.."/"..github_repo_name..".git",
+   branch = (package_version == "dev") and "master" or nil,
+   tag = (package_version ~= "dev") and ("v" .. package_version) or nil,
 }
 
 description = {
    summary = "Lua binding to the iconv",
    detailed = [[
-     Lua binding to the POSIX 'iconv' library, which converts a sequence of
-     characters from one codeset into a sequence of corresponding characters
-     in another codeset.
-   ]],
+      Lua binding to the POSIX 'iconv' library, which converts a sequence of
+      characters from one codeset into a sequence of corresponding characters
+      in another codeset.
+      ]],
    license = "MIT/X11",
-   homepage = "https://github.com/lunarmodules/lua-iconv/"
+   homepage = "https://github.com/"..github_account_name.."/"..github_repo_name,
```

Comment:
most of the lunar modules projects have this. I find it easier to manage since all essential metadata is at the top of the file now. (had too many cases of editing one but forgetting another occurrence).

Anyway, I do get your point, so let me know if the above doesn't change your mind, then I'll update it.

Reply:
Sometimes I like to open these uris from my editor, and this kind of interpolation breaks that. Overall I think this parameterization makes the rockspec harder to read and because it's generated, everything could be hardcoded instead.
But it's a minor annoyance, dont bother. Let's use the general workflow.

Reply:
I run into this a lot with distro packaging, especially in Arch Linux where package manifests are bash scripts and anything *may* be interpreted. I have some editor functions setup to do things like "open the homepage for this package" that process the language and use the output. For example in VIM, this can be done for almost any rockspec, including this one, like so:

```vim
:!sh -c "lua -e 'dofile [[%]] print(description.homepage)' | xdg-open"
```

Obviously the nested quoting is tricky, but with Lua's extra quote types at hand and VIM expanding the % to the current buffer filename, this command when run while editing a rockspec will send the home page to my browser. I have it wrapped in a command that does different things for different file types, but it means I don't have to worry about all URLs being spelled out with no variables in them.

Reply:
nice tip, though the approach may scare contributors away.